cmd/juju/application/bundlediff.go

```diff
@@ -248,11 +249,12 @@ func (c *bundleDiffCommand) bundleDataSource(ctx *cmd.Context) (charm.BundleData
 		return nil, errors.Errorf("couldn't interpret %q as a local or charmstore bundle", c.bundle)
 	}
 
-	dir, err := ioutil.TempDir("", bundleURL.Name)
+	dir, err := ioutil.TempDir("", "bundle-diff-")
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	bundle, err := charmAdaptor.GetBundle(bundleURL, dir)
+	bundlePath := path.Join(dir, bundleURL.Name)
```

Comment:
`s/path/filepath`?. Not 100% sure if path.Join` uses the right delimiter across all OS.

Reply:
No, you're right.